commit_rmn_ocb/plugin.go

```diff
@@ -0,0 +1,152 @@
+package commitrmnocb
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	libocrtypes "github.com/smartcontractkit/libocr/ragep2p/types"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+
+	"github.com/smartcontractkit/chainlink-ccip/commit"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+type Plugin struct {
+	reportingCfg      ocr3types.ReportingPluginConfig
+	nodeID            commontypes.OracleID
+	oracleIDToP2pID   map[commontypes.OracleID]libocrtypes.PeerID
+	log               logger.Logger
+	rmn               Rmn
+	cfg               CommitPluginConfig
+	onChain           OnChain
+	tokenPricesReader reader.TokenPrices
+	ccipReader        reader.CCIP
+	homeChain         reader.HomeChain
+	bgSyncCancelFunc  context.CancelFunc
+	bgSyncWG          *sync.WaitGroup
+}
+
+func NewPlugin(
+	reportingCfg ocr3types.ReportingPluginConfig,
+	nodeID commontypes.OracleID,
+	oracleIDToP2pID map[commontypes.OracleID]libocrtypes.PeerID,
+	log logger.Logger,
+	rmn Rmn,
+	cfg CommitPluginConfig,
+	onChain OnChain,
+	tokenPricesReader reader.TokenPrices,
+	ccipReader reader.CCIP,
+	homeChain reader.HomeChain,
+) *Plugin {
+	p := &Plugin{
+		reportingCfg:      reportingCfg,
+		nodeID:            nodeID,
+		oracleIDToP2pID:   oracleIDToP2pID,
+		log:               log,
+		rmn:               rmn,
+		cfg:               cfg,
+		onChain:           onChain,
+		tokenPricesReader: tokenPricesReader,
+		ccipReader:        ccipReader,
+		homeChain:         homeChain,
+	}
+
+	bgSyncCtx, bgSyncCf := context.WithCancel(context.Background())
+	p.bgSyncCancelFunc = bgSyncCf
+	p.bgSyncWG = &sync.WaitGroup{}
+	p.bgSyncWG.Add(1)
+	commit.BackgroundReaderSync(
+		bgSyncCtx,
+		p.bgSyncWG,
+		log,
+		ccipReader,
+		syncTimeout(cfg.SyncTimeout),
+		time.NewTicker(syncFrequency(p.cfg.SyncFrequency)).C,
+	)
+
+	return p
+}
+
+func (p *Plugin) Close() error {
+	timeout := 10 * time.Second
+	ctx, cf := context.WithTimeout(context.Background(), timeout)
+	defer cf()
+
+	if err := p.ccipReader.Close(ctx); err != nil {
+		return fmt.Errorf("close ccip reader: %w", err)
+	}
+
+	p.bgSyncCancelFunc()
+	p.bgSyncWG.Wait()
+	return nil
+}
+
+func (p *Plugin) decodeOutcome(outcome ocr3types.Outcome) (CommitPluginOutcome, CommitPluginState) {
```

Comment:
Feels like this should return an error instead of just returning `SelectingRangesForReport` if there is an error decoding

Reply:
My logic is: if there's an issue decoding, just go to the "select interval" state instead of erroring-out